sudoku/sudoku.py

```diff
@@ -0,0 +1,147 @@
+# solve sudoku puzzle
+
+
+def solve_board(input_board):
+    """Solve a sudoku puzzle
+
+    Args:
+        input_board (list of list[int]): list of lists with integers
+        representing the sudoku board
+
+    Returns:
+        list of list: solved sudoku board else prints unsolvable board
+    """
+    board = input_board.copy()
+    if solve_helper(board):
+        return board
+    else:
+        print(" Board cannot be solved  ....!")
+
+
+def find_empty_spot(board):
```

Comment:
This is fine, but would be better to look for the next empty.

Since you are already a certain amount of the way through the board, you already know that the previous cells are not going to be empty. Starting at your current coordinates saves some time. Not a huge save since its still O(n) regardless.